### ISSUE TITLE
Problems with browserify

### ISSUE THREAD

**Issue by ekaitz-zarraga** [kripken/sql.js#187]
This kind of environment selectors destroy browserify (code is shortened for brevity).

``` javascript
if (ENVIRONMENT_IS_NODE) {
  var nodeFS = require('fs');
  var nodePath = require('path');
```

Browserify resolves the require even if it's inside of the `if`. Because of this, a big piece of Node.js is Browserified and never used because the code never falls there.

This is a nonsense. Is there any simple workaround to solve this problem inside the library?

It's easy to separate the environment selection in two files, one with the browser part and another with the node part but that's not cool for packaging.

What do you think?

**Comment by lovasoa** [comment]
I think this is a compatibility problem between emscripten and browserify. It should be solved either in one or the other of these projects, shouldn't it ?

**Comment by ekaitz-zarraga** [comment]
I don't know if that part is also automatically generated.

Is it?

**Comment by lovasoa** [comment]
Yes. All the handwritten code is in coffee (https://github.com/kripken/sql.js/tree/master/coffee).

**Comment by ekaitz-zarraga** [comment]
I see.

I would need to make some kind of fork of this then I guess :smile:

**Comment by lovasoa** [comment]
I don't think you need to do so.
See https://github.com/kripken/emscripten/issues/3379 .

**Comment by ekaitz-zarraga** [comment]
Hi!

Thank you very much for the link!

This doesn't solve the problem because it protects the execution time but it still allows Browserify to append all the requires to the final bundle.

I'll try to find a solution with EMSCRIPTEN team.

Thank you.
